packages/list-store/src/useListStore.ts

```diff
@@ -0,0 +1,109 @@
+import {create} from 'zustand';
+import {getUrlWithSearchParams} from './search-params';
+import {SortBy, defaultSortBy} from './sort';
+
+export interface Props {
+  totalCount: number;
+  offset: number;
+  limit: number;
+  query: string;
+  sortBy: SortBy;
+  // Setting newDataNeeded to true will trigger a page reload with new search params
+  newDataNeeded: boolean;
+  initialized: boolean;
+  selectedFilters: {[filterKey: string]: string[] | undefined};
+  setSelectedFilters: (key: string, value: string[]) => void;
+  setSortBy: (sortBy: SortBy) => void;
+  setQuery: (query: string) => void;
+  setPage: (direction: 1 | -1) => void;
+  composed: {
+    urlWithSearchParams: string;
+  };
+  newServerData: ({
```

Comment:
I think it's important context that this setter is not a client action. All other setters are user actions that will result in collecting new data. Both types of actions have different side effects. A user action will also set `offset` and `newDataNeeded`. The `newServerData` will set `isInitialized`.

Somehow I want to make this context clear. Maybe the user actions also need to be renamed. I need some time to think about this.

Reply:
@sdevalk Thinking about the next names. Any thoughts?

User actions:

```
  filterClick
  sortClick / sortChange
  queryChange / queryTyping
  pageChange
```

API change

```
setNewData
```

Reply:
Decided on `[property]Change` for user actions. Because it sets more than just the property, so `set[Property]` can be confusing. And `setNewData` for new API data.